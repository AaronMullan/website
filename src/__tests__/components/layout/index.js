import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery, StaticQuery } from 'gatsby'
import Layout from '../../../components/layout'

const SampleChildren = <p>Sample content</p>

beforeEach(() => {
  const useStateSpy = jest.spyOn(React, 'useState')
  useStateSpy.mockImplementation(init => [init, setState])

  StaticQuery.mockImplementation(({ render }) =>
    render({
      site: {
        siteMetadata: {
          production: true,
        },
      },
    }),
  )

  useStaticQuery.mockImplementation(() => ({
    site: {
      siteMetadata: {
        title: 'Test site',
      },
    },
    allNavigationYaml: {
      edges: [
        {
          node: {
            items: [
              {
                link: '/test-a',
                title: 'Test A',
              },
              {
                link: '/test-b',
                title: 'Test B',
              },
            ],
          },
        },
      ],
    },
  }))
})

describe('Components : Layout : Header', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(() => (
        <Layout title="Sample title">
          <SampleChildren />
        </Layout>
      ))
      .toJSON()
    expect(tree).toMatchSnapshot()

    const treeTextHeavy = renderer
      .create(() => (
        <Layout title="Sample title" textHeavy narrow>
          <SampleChildren />
        </Layout>
      ))
      .toJSON()
    expect(treeTextHeavy).toMatchSnapshot()
  })
})
