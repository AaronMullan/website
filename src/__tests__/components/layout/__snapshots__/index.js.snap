// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Layout : Header renders correctly 1`] = `null`;

exports[`Components : Layout : Header renders correctly 2`] = `null`;
